src/front/glsl/constants.rs

```diff
@@ -201,6 +209,47 @@ impl<'a> ConstantSolver<'a> {
                         let inner = ConstantInner::Scalar { width, value };
                         Ok(self.register_constant(inner, span))
                     }
+                    crate::MathFunction::Clamp => {
+                        let (value, width) = match (const0, const1.unwrap(), const2.unwrap()) {
+                            (
+                                &ConstantInner::Scalar {
+                                    width,
+                                    value: value0,
+                                },
+                                &ConstantInner::Scalar { value: value1, .. },
+                                &ConstantInner::Scalar { value: value2, .. },
+                            ) => (
+                                match (value0, value1, value2) {
+                                    (
+                                        ScalarValue::Sint(a),
+                                        ScalarValue::Sint(b),
+                                        ScalarValue::Sint(c),
+                                    ) => ScalarValue::Sint(a.max(b).min(c)),
+                                    (
+                                        ScalarValue::Uint(a),
+                                        ScalarValue::Uint(b),
+                                        ScalarValue::Uint(c),
+                                    ) => ScalarValue::Uint(a.max(b).min(c)),
+                                    (
+                                        ScalarValue::Float(a),
+                                        ScalarValue::Float(b),
+                                        ScalarValue::Float(c),
+                                    ) => ScalarValue::Float(a.max(b).min(c)),
```

Comment:
nit: Can we add just a comment with something like
```rust
// TODO: when the MSRV is higher or equal to 1.50 switch to f32::clamp
```

Reply:
Yes.

Reply:
Sorry at the time I didn't realize `clamp` panicked on `NaN` so we can't really use it.